### PR TITLE
feat: add `maintain_order` to Expr.unique and Series.unique 

### LINKED ISSUE #1328: enh: add `maintain_order` to Expr.unique and Series.unique
As requested here https://github.com/plotly/plotly.py/pull/4790#issuecomment-2459974589

We already have `maintain_order` for `DataFrame.unique`, my concern is that for some libraries this may turn out to not be guaranteed (and for there to not be any option to guarantee it)

However, a workaround like
```python
token = f'{s.name}_tmp'
s.to_frame().with_row_index(token).unique(col).sort(token).get_column(col)
```
should always be possible, were some library to stop guaranteeing it

It's not clear to me whether PyArrow makes any guarantees - we can always just use it for now, and if it stop making such guarantees, we use the workaround above https://github.com/apache/arrow/issues/44663

### PR DESCRIPTION
**Pull request by raisadz**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #1328

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/_pandas_like/series.py [hunk 1]
```diff
@@ -471,7 +471,12 @@ def abs(self) -> PandasLikeSeries:
     def cum_sum(self) -> PandasLikeSeries:
         return self._from_native_series(self._native_series.cumsum())
 
-    def unique(self) -> PandasLikeSeries:
+    def unique(self, *, maintain_order: bool = False) -> PandasLikeSeries:
+        """
+        NOTE:
+            The param `maintain_order` is only here for compatibility with the Polars API
+            and has no effect on the output.
+        """
```
Comment (FBruzzesi) [inline narwhals/_pandas_like/series.py:479]: Maybe for pandas this is possible via `series.drop_duplicates(keep="first")`?

**Review by FBruzzesi — COMMENTED** [review]
Thanks @raisadz ! I left a comment regarding pandas, which may be doable to guarantee the order to be preserved

File: narwhals/_pandas_like/series.py (hunk 1 above)
Comment (MarcoGorelli) [reply narwhals/_pandas_like/series.py:479]: pandas already maintains order, so I don't think we need to change anything

Comment (FBruzzesi) [reply narwhals/_pandas_like/series.py:479]: Ignore me please 🙈

**Review by MarcoGorelli — APPROVED** [review]
thanks @raisadz (and @FBruzzesi for reviewing)!

**Pull request merged**